sup3r/utilities/era_downloader.py

```diff
@@ -0,0 +1,1016 @@
+"""Download ERA5 file for the given year and month
+
+NOTE: To use this you need to have cdsapi package installed and a ~/.cdsapirc
+file with a url and api key.  Follow the instructions here:
+https://cds.climate.copernicus.eu/api-how-to
+"""
+
+import logging
+import os
+from calendar import monthrange
+from concurrent.futures import (
+    ProcessPoolExecutor,
+    ThreadPoolExecutor,
+    as_completed,
+)
+from glob import glob
+from warnings import warn
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+from netCDF4 import Dataset
+
+from sup3r.utilities.log_interpolator import LogLinInterpolator
+
+logger = logging.getLogger(__name__)
+
+try:
+    import cdsapi
+    c = cdsapi.Client()
+except ImportError as e:
+    msg = f'Could not import cdsapi package. {e}'
+    logger.error(msg)
+
+
+class BaseEraDownloader:
+    """Class to handle downloading ERA5 data"""
+
+    msg = ('To download ERA5 data you need to have a ~/.cdsapirc file '
+           'with a valid url and api key. Follow the instructions here: '
+           'https://cds.climate.copernicus.eu/api-how-to')
+    req_file = os.path.join(os.path.expanduser('~'), '.cdsapirc')
+    assert os.path.exists(req_file), msg
+
+    SFC_VARS = ['10m_u_component_of_wind', '10m_v_component_of_wind',
+                '100m_u_component_of_wind', '100m_v_component_of_wind',
+                'surface_pressure', '2m_temperature', 'geopotential']
+    LEVEL_VARS = ['u_component_of_wind', 'v_component_of_wind',
+                  'geopotential', 'temperature']
+    NAME_MAP = {'u10': 'u_10m', 'v10': 'v_10m', 'u100': 'u_100m',
+                'v100': 'v_100m', 't': 'temperature',
+                't2m': 'temperature_2m', 'u': 'u', 'v': 'v',
+                'sp': 'pressure_0m'}
```

Comment:
this is just a personal thing, i always prefer global lists/dicts to have every entry on a different line. It's a lot easier to see what's there, what they keys are, and easier to add new items.